### PR TITLE
SRE-3134: remediate no-default-service-account found by tfsec

### PR DESCRIPTION
**Pull request by blentz**

### CODE REVIEW

**Comment by tamr-github-integrator** [comment]
Results #1-2 MEDIUM Instance does not have VTPM for shielded VMs enabled. (2 similar results)
────────────────────────────────────────────────────────────────────────────────
  main.tf:43-99
────────────────────────────────────────────────────────────────────────────────
  Individual Causes
  - main.tf:43-99 (google_compute_instance.tamr)
  - main.tf:7-22 (module.minimal)
────────────────────────────────────────────────────────────────────────────────
          ID google-compute-enable-shielded-vm-vtpm
      Impact Unable to prevent unwanted system state modification
  Resolution Enable Shielded VM VTPM

  More Information
  - https://aquasecurity.github.io/tfsec/v1.28.1/checks/google/compute/enable-shielded-vm-vtpm/
  - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#enable_vtpm
────────────────────────────────────────────────────────────────────────────────


Results #3-4 MEDIUM Instance does not have shielded VM integrity monitoring enabled. (2 similar results)
────────────────────────────────────────────────────────────────────────────────
  main.tf:43-99
────────────────────────────────────────────────────────────────────────────────
  Individual Causes
  - main.tf:43-99 (google_compute_instance.tamr)
  - main.tf:7-22 (module.minimal)
────────────────────────────────────────────────────────────────────────────────
          ID google-compute-enable-shielded-vm-im
      Impact No visibility of VM instance boot state.
  Resolution Enable Shielded VM Integrity Monitoring

  More Information
  - https://aquasecurity.github.io/tfsec/v1.28.1/checks/google/compute/enable-shielded-vm-im/
  - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#enable_vtpm
────────────────────────────────────────────────────────────────────────────────


Results #5-6 LOW Instance disk encryption does not use a customer managed key. (2 similar results)
────────────────────────────────────────────────────────────────────────────────
  main.tf:50-56
────────────────────────────────────────────────────────────────────────────────
  Individual Causes
  - main.tf:7-22 (module.minimal)
  - main.tf:43-99 (google_compute_instance.tamr)
────────────────────────────────────────────────────────────────────────────────
          ID google-compute-vm-disk-encryption-customer-key
      Impact Using unmanaged keys does not allow for proper management
  Resolution Use managed keys 

  More Information
  - https://aquasecurity.github.io/tfsec/v1.28.1/checks/google/compute/vm-disk-encryption-customer-key/
  - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#kms_key_self_link
────────────────────────────────────────────────────────────────────────────────


  18 passed, 6 potential problem(s) detected.

**Comment by tamr-github-integrator** [comment]
Results #1-2 MEDIUM Instance does not have VTPM for shielded VMs enabled. (2 similar results)
────────────────────────────────────────────────────────────────────────────────
  main.tf:43-99
────────────────────────────────────────────────────────────────────────────────
  Individual Causes
  - main.tf:43-99 (google_compute_instance.tamr)
  - main.tf:7-22 (module.minimal)
────────────────────────────────────────────────────────────────────────────────
          ID google-compute-enable-shielded-vm-vtpm
      Impact Unable to prevent unwanted system state modification
  Resolution Enable Shielded VM VTPM

  More Information
  - https://aquasecurity.github.io/tfsec/v1.28.1/checks/google/compute/enable-shielded-vm-vtpm/
  - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#enable_vtpm
────────────────────────────────────────────────────────────────────────────────


Results #3-4 MEDIUM Instance does not have shielded VM integrity monitoring enabled. (2 similar results)
────────────────────────────────────────────────────────────────────────────────
  main.tf:43-99
────────────────────────────────────────────────────────────────────────────────
  Individual Causes
  - main.tf:7-22 (module.minimal)
  - main.tf:43-99 (google_compute_instance.tamr)
────────────────────────────────────────────────────────────────────────────────
          ID google-compute-enable-shielded-vm-im
      Impact No visibility of VM instance boot state.
  Resolution Enable Shielded VM Integrity Monitoring

  More Information
  - https://aquasecurity.github.io/tfsec/v1.28.1/checks/google/compute/enable-shielded-vm-im/
  - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#enable_vtpm
────────────────────────────────────────────────────────────────────────────────


Results #5-6 LOW Instance disk encryption does not use a customer managed key. (2 similar results)
────────────────────────────────────────────────────────────────────────────────
  main.tf:50-56
────────────────────────────────────────────────────────────────────────────────
  Individual Causes
  - main.tf:43-99 (google_compute_instance.tamr)
  - main.tf:7-22 (module.minimal)
────────────────────────────────────────────────────────────────────────────────
          ID google-compute-vm-disk-encryption-customer-key
      Impact Using unmanaged keys does not allow for proper management
  Resolution Use managed keys 

  More Information
  - https://aquasecurity.github.io/tfsec/v1.28.1/checks/google/compute/vm-disk-encryption-customer-key/
  - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#kms_key_self_link
────────────────────────────────────────────────────────────────────────────────


  18 passed, 6 potential problem(s) detected.

**Comment by tamr-github-integrator** [comment]
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>tfsec - Results</title>
    <style>
      * { background-color: #222222; color: #ffffff; }
      body { font-size: 20px; } 
      .severity { color: #777777; }
      .CRITICAL { font-weight: bold; color: #cc1111;}
      .HIGH { color: #ff0000; }
      .MEDIUM { color: #ff6600; }
      .LOW { color: #ffff00; }
      table { border: 1px solid #555555; }
      thead tr th {
        background-color: #555555;
        padding: 5px 10px 5px 10px;
      }
      tbody tr td { padding: 5px 10px 5px 10px; }
    </style>
  </head>
  <body>
    <h1>[tfsec] Results</h1>
    <h2>Failed: 6 issue(s)</h2>
    <table class="pure-table">
      <thead>
        <tr><th> # </th><th> ID </th><th> Severity </th><th> Title </th><th> Location </th><th> Description </th></tr>
      </thead>
      <tbody>
        <tr>
          <td>1</td>
          <td><a target="_blank" rel="noopener" href="https://cloud.google.com/security/shielded-cloud/shielded-vm#integrity-monitoring">google-compute-enable-shielded-vm-im</a></td>
          <td class="severity MEDIUM">MEDIUM</td>
          <td><i>Instances should have Shielded VM integrity monitoring enabled</i></td>
          <td><code>main.tf:43-99</code></td>
          <td>Instance does not have shielded VM integrity monitoring enabled.</td>
        </tr>
        <tr>
          <td>2</td>
          <td><a target="_blank" rel="noopener" href="https://cloud.google.com/security/shielded-cloud/shielded-vm#integrity-monitoring">google-compute-enable-shielded-vm-im</a></td>
          <td class="severity MEDIUM">MEDIUM</td>
          <td><i>Instances should have Shielded VM integrity monitoring enabled</i></td>
          <td><code>main.tf:43-99</code></td>
          <td>Instance does not have shielded VM integrity monitoring enabled.</td>
        </tr>
        <tr>
          <td>3</td>
          <td><a target="_blank" rel="noopener" href="https://cloud.google.com/blog/products/identity-security/virtual-trusted-platform-module-for-shielded-vms-security-in-plaintext">google-compute-enable-shielded-vm-vtpm</a></td>
          <td class="severity MEDIUM">MEDIUM</td>
          <td><i>Instances should have Shielded VM VTPM enabled</i></td>
          <td><code>main.tf:43-99</code></td>
          <td>Instance does not have VTPM for shielded VMs enabled.</td>
        </tr>
        <tr>
          <td>4</td>
          <td><a target="_blank" rel="noopener" href="https://cloud.google.com/blog/products/identity-security/virtual-trusted-platform-module-for-shielded-vms-security-in-plaintext">google-compute-enable-shielded-vm-vtpm</a></td>
          <td class="severity MEDIUM">MEDIUM</td>
          <td><i>Instances should have Shielded VM VTPM enabled</i></td>
          <td><code>main.tf:43-99</code></td>
          <td>Instance does not have VTPM for shielded VMs enabled.</td>
        </tr>
        <tr>
          <td>5</td>
          <td><a target="_blank" rel="noopener" href="">google-compute-vm-disk-encryption-customer-key</a></td>
          <td class="severity LOW">LOW</td>
          <td><i>VM disks should be encrypted with Customer Supplied Encryption Keys</i></td>
          <td><code>main.tf:50-56</code></td>
          <td>Instance disk encryption does not use a customer managed key.</td>
        </tr>
        <tr>
          <td>6</td>
          <td><a target="_blank" rel="noopener" href="">google-compute-vm-disk-encryption-customer-key</a></td>
          <td class="severity LOW">LOW</td>
          <td><i>VM disks should be encrypted with Customer Supplied Encryption Keys</i></td>
          <td><code>main.tf:50-56</code></td>
          <td>Instance disk encryption does not use a customer managed key.</td>
        </tr>

      </tbody>
    </table>
  </body>
</html>

**Comment by tamr-github-integrator** [comment]
# [tfsec] Results
## Failed: 6 issue(s)
| # | ID | Severity | Title | Location | Description |
|---|----|----------|-------|----------|-------------|
| 1 | `google-compute-enable-shielded-vm-im` | *MEDIUM* | _Instances should have Shielded VM integrity monitoring enabled_ | `main.tf:43-99` | Instance does not have shielded VM integrity monitoring enabled. |
| 2 | `google-compute-enable-shielded-vm-vtpm` | *MEDIUM* | _Instances should have Shielded VM VTPM enabled_ | `main.tf:43-99` | Instance does not have VTPM for shielded VMs enabled. |
| 3 | `google-compute-vm-disk-encryption-customer-key` | *LOW* | _VM disks should be encrypted with Customer Supplied Encryption Keys_ | `main.tf:50-56` | Instance disk encryption does not use a customer managed key. |
| 4 | `google-compute-enable-shielded-vm-im` | *MEDIUM* | _Instances should have Shielded VM integrity monitoring enabled_ | `main.tf:43-99` | Instance does not have shielded VM integrity monitoring enabled. |
| 5 | `google-compute-enable-shielded-vm-vtpm` | *MEDIUM* | _Instances should have Shielded VM VTPM enabled_ | `main.tf:43-99` | Instance does not have VTPM for shielded VMs enabled. |
| 6 | `google-compute-vm-disk-encryption-customer-key` | *LOW* | _VM disks should be encrypted with Customer Supplied Encryption Keys_ | `main.tf:50-56` | Instance disk encryption does not use a customer managed key. |

File: examples/minimal/variables.tf [hunk 1]
```diff
@@ -18,7 +18,7 @@ variable "zone_id" {
 }
 
 variable "service_account" {
-  default = ""
+  default = "nondefault-service-account@this-project.gserviceaccount.com"
```
[the remark this answers is not in the document]
Comment (blentz) [reply examples/minimal/variables.tf:21]: ```The PR itself is not ready for review. There’s no need to do a PR review on the code itself.```

Was there something unclear about what sort of a review was requested?

**Pull request closed without merging**